### PR TITLE
Add an example of Azure Functions using isolated mode

### PR DESCRIPTION
**Pull request by zhenlan**

### CODE REVIEW

**Comment by zhenlan** [comment]
Once we have the [middleware for Azure Functions](https://github.com/Azure/AppConfiguration-DotnetProvider/issues/287), the example will be updated to use the new package.

File: examples/DotNetCore/AzureFunction/FunctionAppIsolatedMode/AzureAppConfigurationRefreshExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Extensions.Hosting;
+
+namespace FunctionAppIsolatedMode
+{
+    internal static class AzureAppConfigurationRefreshExtensions
+    {
+        public static IFunctionsWorkerApplicationBuilder UseAzureAppConfiguration(this IFunctionsWorkerApplicationBuilder builder)
+        {
+            if (builder == null)
```
Comment (jimmyca15) [inline examples/DotNetCore/AzureFunction/FunctionAppIsolatedMode/AzureAppConfigurationRefreshExtensions.cs:11]: I don't expect argument validation on the extended object an extension method considering it's usage.

File: examples/DotNetCore/AzureFunction/FunctionAppIsolatedMode/AzureAppConfigurationRefreshMiddleware.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+using System.Threading.Tasks;
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Azure.Functions.Worker.Middleware;
+using Microsoft.Extensions.Configuration.AzureAppConfiguration;
+
+namespace FunctionAppIsolatedMode
+{
+    internal class AzureAppConfigurationRefreshMiddleware : IFunctionsWorkerMiddleware
+    {
+        public IEnumerable<IConfigurationRefresher> Refreshers { get; }
```
Comment (jimmyca15) [inline examples/DotNetCore/AzureFunction/FunctionAppIsolatedMode/AzureAppConfigurationRefreshMiddleware.cs:11]: Why is Refreshers public?

**Review by jimmyca15 — APPROVED** [review]

File: examples/DotNetCore/AzureFunction/FunctionAppIsolatedMode/AzureAppConfigurationRefreshExtensions.cs (hunk 1 above)
Comment (zhenlan) [reply examples/DotNetCore/AzureFunction/FunctionAppIsolatedMode/AzureAppConfigurationRefreshExtensions.cs]: Removed

File: examples/DotNetCore/AzureFunction/FunctionAppIsolatedMode/AzureAppConfigurationRefreshMiddleware.cs (hunk 2 above)
Comment (zhenlan) [reply examples/DotNetCore/AzureFunction/FunctionAppIsolatedMode/AzureAppConfigurationRefreshMiddleware.cs]: No reason. I copied the code from the existing asp.net core middleware :). Updated to use private field.

**Pull request merged**